SPECS/kernel/kernel.spec

```diff
@@ -35,9 +35,11 @@ BuildRequires:  bc
 BuildRequires:  diffutils
 BuildRequires:  dwarves
 BuildRequires:  elfutils-libelf-devel
+BuildRequires:  gettext
 BuildRequires:  glib-devel
 BuildRequires:  kbd
 BuildRequires:  kmod-devel
+BuildRequires:  libcap-devel
```

Comment:
Thanks for taking the time to review Mariner! I verfied these are missing and found a few more. Could you update the PR to also include these additional BRs?

glibc-devel
```
scripts/basic/fixdep.c:92:10: fatal error: sys/types.h: No such file or directory
  92 | #include <sys/types.h>
     |          ^~~~~~~~~~~~~
compilation terminated.
make[1]: *** [scripts/Makefile.host:95: scripts/basic/fixdep] Error 1
make: *** [Makefile:555: scripts_basic] Error 2
```
binutils
```
gcc: fatal error: cannot execute 'as': execvp: No such file or directory
compilation terminated.
make[1]: *** [scripts/Makefile.host:95: scripts/basic/fixdep] Error 1
make: *** [Makefile:555: scripts_basic] Error 2
error: Bad exit status from /var/tmp/rpm-tmp.ETozJ7 (%prep)
```
Kernel-headers
```
/usr/include/bits/local_lim.h:38:10: fatal error: linux/limits.h: No such file or directory
  38 | #include <linux/limits.h>
     |          ^~~~~~~~~~~~~~~~
```
flex
```
/bin/sh: line 1: flex: command not found
make[1]: *** [scripts/Makefile.host:9: scripts/kconfig/lexer.lex.c] Error 127
```
bison
```
/bin/sh: line 1: bison: command not found
make[1]: *** [scripts/Makefile.host:17: scripts/kconfig/parser.tab.h] Error 127
```

repro steps
```
docker run -it mcr.microsoft.com/cbl-mariner/base/core:2.0 /bin/bash
tdnf install wget rpmbuild rpmdevtools
wget <kernel srpm url>
rpmdev-setuptree
rpm2cpio kernel.src.rpm | cpio -idmv
rpmbuild -ba SPECS/kernel.spec
```

Reply:
> Thanks for taking the time to review Mariner! I verfied these are missing and found a few more. Could you update the PR to also include these additional BRs? I also included the error messages for context
> 
> glibc-devel

it is my understanding that `glibc-devel` and `binutils` ought not generally be listed as a `BuildRequires` in fedora-derived distros, since the compiler packages depend on them. insted, this kernel specfile ought IMHO require `gcc`, which will drag in "all packages necessary for C development".

`flex` and `bison` are buildrequires of gcc, but not requires, so they ought indeed be added.

i've gone ahead and updated my PR with `flex`, `bison`, and `gcc`. I believe this to be correct, but let me know if i've overlooked something. i don't believe there to be any c++ in the kernel sources, but if so, we'd want `gcc-g++` instead of `gcc`.

https://docs.fedoraproject.org/en-US/packaging-guidelines/C_and_C++

Reply:
It appears I missed `kernel-headers` in my inital comment, here is the error message

```
/usr/include/bits/local_lim.h:38:10: fatal error: linux/limits.h: No such file or directory
   38 | #include <linux/limits.h>
      |          ^~~~~~~~~~~~~~~~
```

After taking a second look at these BRs, we should use `build-essential`  to encapsulate some of these (see the spec [here](https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/build-essential/build-essential.spec) ). Note that your original BRs `gettext` , `libcap-devel` and `flex`

Just some notes here, we are technically not fedora based although we do have a large number of specs that follow their format. For instance, Fedora does not have a build-essential package (whereas other distros such as Ubuntu does).

Reply:
I've added build-essential and removed the items it provides (gcc and bison).

I'll keep in mind that you're using RPM but not fedora-derived, thanks!

PTAL.